src/Database.ts

```diff
@@ -365,6 +365,15 @@ export class Database<
     await migrate(this, config)
   }
 
+  /**
+   * Performs a database migration read operation.
+   * @returns List of objects with name, id of migrations and SQL code for up and down migration
+   * You can run up and down with `db.exec(migrations[0].up)` or `db.exec(migrations[0].down)`. Or with other commands that can run SQL code.
+   */
+  readMigrations (migrationPath?: string): Promise<IMigrate.MigrationData[]> {
```

Comment:
Is the intent to just read a list of migrations and then execute them?

It almost feels like this should belong in its own Migrations class and the constructor would accept a Database instance 

then you can have a method for just reading the list of migrations / status

and then have another method(s) for executing up / down based on the list of migrations passed through

Reply:
Yes, I understand that it's better. But it'll take some time. So let's me finish my tasks and I plan to make this changes on the next week

Changes:
- add Migration class
- add run, show, reverse migration functions
- add a commander to run these functions to look on the migrations
- add config options where to look for Database instance or config options. With default behaviour. But I plan to add config as a parameter in the command line

WDYT?

Reply:
The overall idea sounds good. I'd have to see the impl / interface to have additional thoughts

Thanks for doing this. We could also probably move the migration logic to the migration class too.

The Database instance can still keep the `migrate` method for backwards-compat but would call the new Migration class

Reply:
Hi, I replied a little later but I have time to make changes)